### PR TITLE
Added index, completion and references for table names, column names …

### PR DESCRIPTION
**Pull request by bohdan-harniuk**
**Description (*)**
This PR adds indexes, completion, references, test coverage on completion and references on table and column names in the db_schema.xml file.

image
image
image

**Also, was changed the assertHasReferenceToXmlTag method of the BaseReferenceTestCase class to work with PolyVariantReferenceBase type that retrieves multiple references.**
image

**After that change all tests passed successfully:**
image

**Contribution checklist** (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with integration/functional tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by VitaliyBoyko — CHANGES_REQUESTED** [review]

File: src/com/magento/idea/magento2plugin/indexes/IndexManager.java [hunk 1]
```diff
@@ -41,6 +42,7 @@ public static void manualReindex() {
             // xml|di configuration
             PluginIndex.KEY,
             VirtualTypeIndex.KEY,
+            TableAndColumnNameIndex.KEY,
```
[the remark this answers is not in the document]
Comment (bohdan-harniuk) [reply src/com/magento/idea/magento2plugin/indexes/IndexManager.java]: Done

**Review by VitaliyBoyko — CHANGES_REQUESTED** [review]
image
An element should not have a reference to itself ⛔

**Comment by bohdan-harniuk** [comment]
Fixed "An element should not have a reference to itself".

**Review by VitaliyBoyko — APPROVED** [review]
QA passed ✅ 
image
image

**Pull request merged**